packages/core/contracts/governance/MetaHumanGovernor.sol

```diff
@@ -0,0 +1,572 @@
+pragma solidity ^0.8.9;
+
+import '@openzeppelin/contracts/governance/Governor.sol';
+import '@openzeppelin/contracts/governance/extensions/GovernorSettings.sol';
+import '@openzeppelin/contracts/governance/extensions/GovernorVotes.sol';
+import '@openzeppelin/contracts/governance/extensions/GovernorVotesQuorumFraction.sol';
+import '@openzeppelin/contracts/governance/extensions/GovernorTimelockControl.sol';
+import './CrossChainGovernorCountingSimple.sol';
+import './DAOSpokeContract.sol';
+import './wormhole/IWormholeRelayer.sol';
+import './wormhole/IWormholeReceiver.sol';
+import './magistrate/Magistrate.sol';
+import 'hardhat/console.sol';
+
+/**
+ * @title MetaHumanGovernor
+ *   @dev MetaHumanGovernor is a contract that serves as a governance system for MetaHuman-related operations. It extends multiple contracts to incorporate various functionalities.
+ *
+ *   This is the hub contract in the MetaHuman governance smart contracts infrastructure.
+ *   It's based on OpenZeppelin's {Governor} implementation along with basic extensions.
+ *   For more details check out [OpenZeppelin's documentation](https://docs.openzeppelin.com/contracts/4.x/api/governance#governor).
+ */
+contract MetaHumanGovernor is
+    Governor,
+    GovernorSettings,
+    CrossChainGovernorCountingSimple,
+    GovernorVotes,
+    GovernorVotesQuorumFraction,
+    GovernorTimelockControl,
+    Magistrate,
+    IWormholeReceiver
+{
+    IWormholeRelayer public immutable wormholeRelayer;
+    uint256 internal constant GAS_LIMIT = 500_000;
+    uint256 public immutable secondsPerBlock;
+    uint16 public immutable chainId;
+
+    mapping(bytes32 => bool) public processedMessages;
+    mapping(uint256 => bool) public collectionStarted;
+    mapping(uint256 => bool) public collectionFinished;
+
+    /**
+     * @dev Contract constructor.
+     *  @param _token The address of the token contract used for voting.
+     *  @param _timelock The address of the timelock contract used for delayed execution.
+     *  @param _spokeContracts An array of CrossChainAddress structs representing the spoke contracts.
+     *  @param _chainId The chain ID of the current contract.
+     *  @param _wormholeRelayerAddress The address of the wormhole automatic relayer contract used for cross-chain communication.
+     */
+    constructor(
+        IVotes _token,
+        TimelockController _timelock,
+        CrossChainAddress[] memory _spokeContracts,
+        uint16 _chainId,
+        address _wormholeRelayerAddress,
+        address _magistrateAddress,
+        uint256 _secondsPerBlock
+    )
+        Governor('MetaHumanGovernor')
+        GovernorSettings(
+            1,
+            /* 1 block */ 20 * 15,
+            /* 20 blocks per minute * 15 minutes (polygon mumbai) */ 0
+        ) //TODO:prod in production voting delay, voting period, proposal threshold needs to be changed to value of choice. Depending on block time on selected hub chain and desired period
```

Comment:
Can't we make these governor settings constructor params of MetaHumanGovernor contract? I don't think it's good to modify values directly for production deployment.

Reply:
prettier is blocking the commit when running :` yarn format:contracts      `

Reply:
Fixed with @portuu3 .